pyartcd/pyartcd/pipelines/prepare_release.py

```diff
@@ -759,21 +762,30 @@ def send_notification_email(self, advisories: Dict[str, int], jira_link: str):
 @click_coroutine
 async def prepare_release(runtime: Runtime, group: str, assembly: str, name: Optional[str], date: str,
                           package_owner: Optional[str], nightlies: Tuple[str, ...], default_advisories: bool, include_shipped: bool):
-    # parse environment variables for credentials
-    jira_token = os.environ.get("JIRA_TOKEN")
-    if not runtime.dry_run and not jira_token:
-        raise ValueError("JIRA_TOKEN environment variable is not set")
-    # start pipeline
-    pipeline = PrepareReleasePipeline(
-        runtime=runtime,
-        group=group,
-        assembly=assembly,
-        name=name,
-        date=date,
-        nightlies=nightlies,
-        package_owner=package_owner,
-        jira_token=jira_token,
-        default_advisories=default_advisories,
-        include_shipped=include_shipped,
-    )
-    await pipeline.run()
+    slack_client = runtime.new_slack_client()
+    slack_client.bind_channel(assembly)
+    await slack_client.say(f":construction: prepare-release for {name if name else assembly} :construction:")
+    try:
+        # parse environment variables for credentials
+        jira_token = os.environ.get("JIRA_TOKEN")
+        if not runtime.dry_run and not jira_token:
+            raise ValueError("JIRA_TOKEN environment variable is not set")
+        # start pipeline
+        pipeline = PrepareReleasePipeline(
+            slack_client=slack_client,
+            runtime=runtime,
+            group=group,
+            assembly=assembly,
+            name=name,
+            date=date,
+            nightlies=nightlies,
+            package_owner=package_owner,
+            jira_token=jira_token,
+            default_advisories=default_advisories,
+            include_shipped=include_shipped,
+        )
+        await pipeline.run()
+        await slack_client.say(f":white_check_mark: prepare-release for {name if name else assembly} completes.")
+    except Exception as e:
```

Comment:
I think this would catch ValueError at line 772, thus preventing the build from turning red. Is there a specific reason to catch broad exceptions now?

Reply:
line 772 is also a kind of exception, I think the pipeline won't run, and build job will be red if the code goes to 772, cat exception on the top level is to make our error slack alert include all possible failures in pipeline as we did in jenkinsfile finally segment

Reply:
And I think for other job with slack alert should also have such design instead of init slack client in pipeline.run() because exception may also happened in init step.

Reply:
This code:
```
    try:
        raise ValueError()
    except Exception:
        print('all good!')
```
will not raise an exception. You're catching everything, thus the pipeline won't run but the build will not be red. 

It's ok to initialize slack before the pipeline runs; what I think we shouldn't do is catch any exception randomly. Being more specific in scope and exception type is desirable, otherwise we might not notice when things break

Reply:
check line 791, it will forward the error to Jenkins and make build red like https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/hack/job/ximhan/job/build%252Fprepare-release/6/consoleFull

Reply:
to catch and through again here is for checking overall status then said "the job failed" otherwise message needs to be append under each exception